### PR TITLE
Use the Paper FeedbackForwardingSender for Discord to MC commands if it exists

### PR DESCRIPTION
**Pull request by sub-kek**
I noticed a long time ago that kyori components messages are not sent to discord. I found a simple solution inside Paper Server Implementation.

### CODE REVIEW

**Comment by sub-kek** [comment]
Perhaps we should create a separate class for sendToDiscord(String message) because it is a duplicate of the method from CommandSenderDynamic.

File: src/main/java/github/scarsz/discordsrv/util/FeedbackCommandSenderUtil.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    }
+
+    private CommandSender createCommandSender() {
+        try {
+            Class<?> serverClass = plugin.getServer().getClass();
+            Method createCommandSenderMethod = serverClass.getMethod("createCommandSender", Consumer.class);
+            Consumer<Object> serializerFunction = this::processComponent;
+            return (CommandSender) createCommandSenderMethod.invoke(plugin.getServer(), serializerFunction);
+        } catch (Throwable e) {
+            plugin.getLogger().severe("Error create commandSender: " + e.getMessage());
```
Comment (Dinty1) [inline src/main/java/github/scarsz/discordsrv/util/FeedbackCommandSenderUtil.java:49]:
```suggestion
            plugin.getLogger().severe("Error creating commandSender: " + e.getMessage());
```

File: src/main/java/github/scarsz/discordsrv/util/FeedbackCommandSenderUtil.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            Class<?> componentInterface = Class.forName(dot("net{}kyori{}adventure{}text{}Component"));
+
+            Method plainTextMethod = serializerInterface.getMethod("plainText");
+            Method serializeMethod = serializerInterface.getMethod("serialize", componentInterface);
+
+            Object serializerInstance = plainTextMethod.invoke(null);
+
+            sendToDiscord((String) serializeMethod.invoke(serializerInstance, component));
+        } catch (Throwable e) {
+            plugin.getLogger().severe("Error serialize not relocated component to String: " + e.getMessage());
```
Comment (Dinty1) [inline src/main/java/github/scarsz/discordsrv/util/FeedbackCommandSenderUtil.java:66]:
```suggestion
            plugin.getLogger().severe("Error serializing not relocated component to String: " + e.getMessage());
```

**Review by FlingisenFlong — APPROVED** [review]

**Review by Dinty1 — CHANGES_REQUESTED** [review]
Seems to work fine. Just a couple of grammar nitpicks

**Comment by sub-kek** [comment]
Perhaps add a configuration option for this. Since some plugins found only luckperms throws an error when using this sender.

**Review by Vankka — CHANGES_REQUESTED** [review]
> Perhaps add a configuration option for this. Since some plugins found only luckperms throws an error when using this sender.

Such errors should be reported to the erroring plugins, rather than degrading functionality

**Comment by sub-kek** [comment]
> Such errors should be reported to the erroring plugins, rather than degrading functionality

Fair enough

**Comment by granny** [comment]
Why was this closed?

**Comment by sub-kek** [comment]
> Why was this closed?

Sorry my mistake (force push). Can you reopen this?

**Comment by Dinty1** [comment]
Your force push seems to have made it think there are no commits to be pulled. Might be better to make a new one (and ensure you didn’t accidentally wipe other parts of the commit history)

**Comment by sub-kek** [comment]
> Your force push seems to have made it think there are no commits to be pulled. Might be better to make a new one (and ensure you didn’t accidentally wipe other parts of the commit history)

I know how to fix it anyway I have everything in the local repository.

**Comment by granny** [comment]
If you force push again it should make it possible to reopen the PR

**Comment by sub-kek** [comment]
https://github.com/DiscordSRV/DiscordSRV/pull/1727

**Pull request closed without merging**